scripts/HistFitter.py

```diff
@@ -222,9 +223,44 @@ def GenerateFitAndPlotCPP(fc, anaName, drawBeforeFit, drawAfterFit, drawStackPlo
         else:
             log.error("Install the pyhf command line tool by running 'pip install pyhf'.")
             sys.exit()
-        #Import modules and set up backend
+        #Import modules
         import usepyhf
-        pyhf.set_backend(pyhf.tensorlib, pyhf.optimize.minuit_optimizer(tolerance=1e-3))
+        #Set up backend
+        if HistFitterArgs.pyhf_backend=="numpy":
+            try:
+                import numpy
+                log.info("import numpy successful. Setting it as pyhf backend.")
+                pyhf.set_backend("numpy", custom_optimizer=pyhf.optimize.minuit_optimizer(tolerance=1e-3))
+            except ImportError:
+                log.error("import numpy failed. Install the python numpy module by running 'pip install numpy'.")
+                sys.exit()
+        elif HistFitterArgs.pyhf_backend=="tensorflow":
+            try:
+                import tensorflow
+                log.info("import tensorflow successful. Setting it as pyhf backend.")
+                pyhf.set_backend("tensorflow", custom_optimizer=pyhf.optimize.minuit_optimizer(tolerance=1e-3))
+            except ImportError:
+                log.error("import tensorflow failed. Install the python tensorflow module by running 'pip install tensorflow'.")
+                sys.exit()
+        elif HistFitterArgs.pyhf_backend=="pytorch":
+            try:
+                import pytorch
+                log.info("import pytorch successful. Setting it as pyhf backend.")
+                pyhf.set_backend("pytorch", custom_optimizer=pyhf.optimize.minuit_optimizer(tolerance=1e-3))
+            except ImportError:
+                log.error("import pytorch failed. Install the python pytorch module by running 'pip install pytorch'.")
+                sys.exit()
+        elif HistFitterArgs.pyhf_backend=="jax":
+            try:
+                import jax
+                log.info("import jax successful. Setting it as pyhf backend.")
+                pyhf.set_backend("jax", custom_optimizer=pyhf.optimize.minuit_optimizer(tolerance=1e-3))
+            except ImportError:
+                log.error("import jax failed. Install the python jax module by running 'pip install jax'.")
+                sys.exit()
+
+            
+        #pyhf.set_backend(custom_optimizer=pyhf.optimize.minuit_optimizer(tolerance=1e-3))
```

Comment:
This code seems overly complicated. Why not let `pyhf` handle it? Otherwise, you're now tying the backends available in this code as well as in the `pyhf` code, and there's no guarantee all of these backends will exist or if `pyhf` will have additional backends.

In any case, if you try to set a backend that cannot be imported, `pyhf` will crash for you, so there's no need to do these checks. You could just simplify as:

```py
pyhf.set_backend(HistFitterArgs.pyhf_backend, custom_optimizer=pyhf.optimize.minuit_optimizer(tolerance=1e-3))
```

Reply:
Ok, I wanted it to be obvious for the user/developer what was failing, but I can do that :) Btw, Giordon, do you know how I can inspect more thoroughly what is going wrong in the pytest? It is passing in my setup, so I am guessing that it is related to some library/path not working correctly.

Reply:
Nevermind, I see where it is! Still the directory issue.